### ISSUE TITLE
Ansible plays end with Exception TypeError: 'NoneType' object is not callable

### ISSUE THREAD

**Issue by ashemedai** [ansible/ansible#30755]
##### ISSUE TYPE
Bug Report

##### COMPONENT NAME
core

##### ANSIBLE VERSION

```
ansible 2.4.0.0
  config file = None
  configured module search path = [u'/home/jeroenr/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.13+ (default, Jul 19 2017, 18:15:03) [GCC 6.4.0 20170704]

```

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

Debian GNU/Linux 8.9 (jessie)

##### SUMMARY

Every playbook run ends with:
Exception TypeError: "'NoneType' object is not callable" in <function remove at 0x7fd27fcf5b18> ignored

##### STEPS TO REPRODUCE

Run any available playbook with ansible-playbok

##### EXPECTED RESULTS

Playbook ends without above exception being thrown.

##### ACTUAL RESULTS

Playbook ends with:
Exception TypeError: "'NoneType' object is not callable" in <function remove at 0x7fd27fcf5b18> ignored

Apparently this is due to a missing change in Python 2.7, which is present in Python 3.5.4 and onwards:
https://github.com/python/cpython/commit/9cd7e17640a49635d1c1f8c2989578a8fc2c1de6

Applying said patch to my local 2.7 tree makes the exception reporting go away.

Logging this ticket both for other people to find it and discussion on how to best deal with cases like this. We currently still use Python 2.7 with Ansible due to 3 not being fully supported with past releases.

**Comment by abadger** [comment]
Interesting.  I had only seen this bug in Python-3.5.  I don't see the patch in my distribution's Python-2.7.13 package but perhaps they have a different patch that affects it.

https://bugs.python.org/issue29519#msg299550 <=== This implies that the bug will be fixed in Python-2.7.14 as well.

**Comment by bcoca** [comment]
I run 2.7.13 and have never seen this output, only when running 3.5.3

**Comment by ashemedai** [comment]
That's indeed interesting. I can try it against a pristine installation of Debian 9 on a VirtualBox if that helps?

**Comment by abadger** [comment]
Sure.  Always helpful to have more information.  The upstream bug does say
they are applying the patch to the next version of python-2.7.x so it seems
like the errors could potentially be there too... we've jsut never seen it
for some reason or another.

**Comment by ashemedai** [comment]
OK, got some data points. This system I am using was tracking sid for a while and recently I set it to buster. The Python 2.7 package I still have installed is 2.7.13-4 from sid. According to http://metadata.ftp-master.debian.org/changelogs/main/p/python2.7/python2.7_2.7.14-2_changelog the 2.7.13-4 synced with Python 2.7's repository. 

Did some more digging and in Python 2.7.14, which got released on the 16th, it does contain this weakref.py fix.

So my case seems to be one of having been caught in between package updates. No need for any other action it seems. So I'll just close this one, apologies for the noise.
